File: java-grpc-spring-boot-module/grpc-server/src/main/java/com/binghe/HelloGrpcService.java
package com.binghe;

import com.binghe.proto.HelloRequest;
import com.binghe.proto.HelloResponse;
import com.binghe.proto.HelloServiceGrpc;
import io.grpc.stub.StreamObserver;
import net.devh.boot.grpc.server.service.GrpcService;

@GrpcService
public class HelloGrpcService extends HelloServiceGrpc.HelloServiceImplBase {

    @Override
    public void hello(HelloRequest request, StreamObserver<HelloResponse> responseObserver) {
        String greeting = "Hello" + request.getFirstName() + "," + request.getLastName();
        HelloResponse response = HelloResponse.newBuilder()
                .setGreeting(greeting)
                .build();
        responseObserver.onNext(response);
        responseObserver.onCompleted();
    }

    @Override
    public void helloServerStream(HelloRequest request, StreamObserver<HelloResponse> responseObserver) {
        super.helloServerStream(request, responseObserver);
    }

    @Override
    public StreamObserver<HelloRequest> helloClientStream(StreamObserver<HelloResponse> responseObserver) {
        return super.helloClientStream(responseObserver);
    }

    @Override
    public StreamObserver<HelloRequest> helloBiStream(StreamObserver<HelloResponse> responseObserver) {
        return super.helloBiStream(responseObserver);
    }
}

File: kotlin-grpc-coroutine-spring-boot/grpc-server/src/main/kotlin/com/binghe/hello/HelloGrpcService.kt
package com.binghe.com.binghe.hello


import com.binghe.proto.HelloRequest
import com.binghe.proto.HelloResponse
import com.binghe.proto.HelloServiceGrpcKt.HelloServiceCoroutineImplBase
import org.lognet.springboot.grpc.GRpcService

@GRpcService
class HelloGrpcService : HelloServiceCoroutineImplBase() {

    override suspend fun hello(request: HelloRequest): HelloResponse {
        val response = "Hello ${request.firstName} ${request.lastName}"
        println("HelloWorldService.hello called. $response")
        return HelloResponse.newBuilder()
            .setGreeting(response)
            .build()
    }
}